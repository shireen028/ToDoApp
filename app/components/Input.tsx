import React, {Component} from 'react';
import {StyleSheet, View, TextInput} from 'react-native';

export class Input extends Component {
  constructor(props: any) {
    super(props);
    this.state = {
      borderColor: '#F2EBDD',
    };
  }
  onFocus = () => {
    this.setState({borderColor: 'blue'});
  };
  onBlur = () => {
    this.setState({borderColor: '#F2EBDD'});
  };
  render() {
    return (
      <View style={[styles.container, {borderColor: this.state.borderColor}]}>
        <TextInput
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          {...this.props}
          style={styles.input}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    backgroundColor: 'white',
    borderRadius: 8,
    paddingHorizontal: 20,
    paddingVertical: 15,
    marginVertical: 12,
    marginLeft: 10,
    marginRight: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    height: 100,
  },
  input: {
    fontSize: 16,
    minHeight: 100,
    width: '98%',
    color: 'black'
  },
});
