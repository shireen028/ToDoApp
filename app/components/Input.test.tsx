import React from 'react';
import 'react-native';
import {Input} from './Input';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Input placeholder="Message" />).toJSON();
  expect(tree).toMatchSnapshot();
});
