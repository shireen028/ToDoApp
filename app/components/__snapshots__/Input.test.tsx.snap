// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "justifyContent": "space-between",
        "marginLeft": 10,
        "marginRight": 10,
        "marginVertical": 12,
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      },
      Object {
        "borderColor": "#F2EBDD",
      },
    ]
  }
>
  <TextInput
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Message"
    style={
      Object {
        "color": "black",
        "fontSize": 16,
        "minHeight": 100,
        "width": "98%",
      }
    }
  />
</View>
`;
